prow/github/client.go

```diff
@@ -3582,7 +3582,8 @@ func (c *client) EnsureFork(forkingUser, org, repo string) (string, error) {
 	if err != nil {
 		return repo, fmt.Errorf("could not fetch all existing repos: %v", err)
 	}
-	if !repoExists(fork, repos) {
+	// if the repo does not exist, or it does, but is not a fork of the repo we want
+	if exists, response := repoExists(fork, repos); !exists || response.Parent.FullName != fmt.Sprintf("%s/%s", org, repo) {
```

Comment:
`repoExists` doesn't need to return two variables. One possible alternative is:

```
func getFork(fork string, repos []Repo) *Repo {
  // If not found, return nil
  // return pointer of the repo
}
```

Then this line can be:

```
if repo := getFork(fork, repos); repo == nil || repo.Parent.FullName != fmt.Sprintf("%s/%s", org, repo) {
```

Reply:
Done